File: algorithms/Armstrong Number/armstrong.java
/* #this is a program to check armstrong number */
import java.util.*;
class armstrong()
{
    public static void main()
    {
        Scanner sc=new Scanner (System.in);
        int num=sc.nextInt();
        /* to seperate the numbers into digits */
        temp=num;
        while(temp>0)
        {
            d=temp%10;
            temp=temp/10;
            sum=sum+ d*d*d;
        }
        if(sum==num)
        {
            System.out.println("Number is Armstrong");
        }
        else
        System.out.println("Number is not an Armstrong number");
    }
}

File: algorithms/Armstrong Number/Armstrong.cpp
#include <cmath>
#include <iostream>
using namespace std;

bool armstrong(int n)
{
    int temp=n,no_of_digits=0,sum=0; // Initial Variables
    
    // Logic to count number of digits in temp
    while(temp!=0)
    {
        temp/=10;
        no_of_digits++;
    }
    temp = n;
   // Logic to Store the sum

    while(temp!=0)
    {
        sum+=pow(temp%10,no_of_digits);  // Adding digit raised to power in sum 
        temp/=10;  // Integer Division by 10 
    }
    return sum==n;

}

// Driver Code
int main()
{
    int x = 371;
    
    if(armstrong(x))
    cout<<x<<" is an Armstrong Number";
    else
    cout<<x<<" is not an Armstrong Number";
}